pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test: TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr failed

pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test.TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr [failed](https://mesolite.cluster.engflow.com/invocations/default/4d8d75ae-25bb-4f84-a673-b74cc9e94617?testReportRun=1&testReportShard=16&testReportAttempt=1#targets-Ly9wa2cvY2NsL3NjaGVtYWNoYW5nZXJjY2wvc2N0ZXN0YmFja3VwY2NsOnNjdGVzdGJhY2t1cGNjbF90ZXN0) on release-25.3 @ [02c9578b4f48443d53a886e9399960878ae3c95a](https://github.com/cockroachdb/cockroach/commits/02c9578b4f48443d53a886e9399960878ae3c95a):


```
google.golang.org/grpc/internal/transport.(*recvBufferReader).read(0xc010055d10, {0xc00d36a730, 0x5, 0x5})
	external/org_golang_google_grpc/internal/transport/transport.go:183 +0x8a
google.golang.org/grpc/internal/transport.(*recvBufferReader).Read(0xc010055d10, {0xc00d36a730?, 0xc011ea9020?, 0xc00868ba98?})
	external/org_golang_google_grpc/internal/transport/transport.go:177 +0x15b
google.golang.org/grpc/internal/transport.(*transportReader).Read(0xc00d36a690, {0xc00d36a730?, 0xc00868bb10?, 0xc995a5?})
	external/org_golang_google_grpc/internal/transport/transport.go:514 +0x2c
io.ReadAtLeast({0x84003e0, 0xc00d36a690}, {0xc00d36a730, 0x5, 0x5}, 0x5)
	GOROOT/src/io/io.go:335 +0x90
io.ReadFull(...)
	GOROOT/src/io/io.go:354
google.golang.org/grpc/internal/transport.(*Stream).Read(0xc00eb24360, {0xc00d36a730, 0x5, 0x5})
	external/org_golang_google_grpc/internal/transport/transport.go:498 +0x96
google.golang.org/grpc.(*parser).recvMsg(0xc00d36a720, 0x7fffffff)
	external/org_golang_google_grpc/rpc_util.go:600 +0x46
google.golang.org/grpc.recvAndDecompress(0xc00a1d02e4?, 0xc00eb24360, {0x0, 0x0}, 0x7fffffff, 0x0, {0x84335f0, 0xd14fc00})
	external/org_golang_google_grpc/rpc_util.go:730 +0x5a
google.golang.org/grpc.recv(0xc00d36a720, {0x7fca95dcbf50, 0xc00172ccc0}, 0xc00eb24360, {0x0, 0x0}, {0x71edbe0, 0xc00f928060}, 0x7fffffff, 0x0, ...)
	external/org_golang_google_grpc/rpc_util.go:796 +0xa7
google.golang.org/grpc.(*serverStream).RecvMsg(0xc00e985c20, {0x71edbe0, 0xc00f928060})
	external/org_golang_google_grpc/stream.go:1712 +0x147
github.com/cockroachdb/cockroach/pkg/kv/kvserver/closedts/ctpb.(*sideTransportPushUpdatesServer).Recv(0xc01123b3a0)
	bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/closedts/ctpb/ctpb_go_proto_/github.com/cockroachdb/cockroach/pkg/kv/kvserver/closedts/ctpb/service.pb.go:484 +0x46
github.com/cockroachdb/cockroach/pkg/kv/kvserver/closedts/sidetransport.(*incomingStream).Run.func1({0x844fca0?, 0xc005538460?})
	pkg/kv/kvserver/closedts/sidetransport/receiver.go:317 +0xc2
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func1({0x844fca0, 0xc005538460})
	pkg/util/stop/stopper.go:438 +0x8a
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 1986387
	pkg/util/stop/stopper.go:436 +0x137

goroutine 1984532 [select]:
github.com/cockroachdb/cockroach/pkg/rpc.(*pooledStream[...]).runOnce(0x84acbc0, {0x844f280, 0xc00f4d2640})
	pkg/rpc/stream_pool.go:107 +0xea
github.com/cockroachdb/cockroach/pkg/rpc.(*pooledStream[...]).run(0x84acbc0, {0x844f280, 0xc00f4d2640})
	pkg/rpc/stream_pool.go:102 +0x7b
github.com/cockroachdb/cockroach/pkg/rpc.(*streamPool[...]).newPooledStream.func2()
	pkg/rpc/stream_pool.go:314 +0xa5
created by github.com/cockroachdb/cockroach/pkg/rpc.(*streamPool[...]).newPooledStream in goroutine 1984504
	pkg/rpc/stream_pool.go:312 +0x33f

goroutine 1982612 [select]:
github.com/cockroachdb/cockroach/pkg/kv/kvclient/rangefeed.(*RangeFeed).processEvents(0xc00a77bd40, {0x844f280, 0xc010055ae0}, {0x84aec00, 0xc0100552c0}, 0xc00b9e4540)
	pkg/kv/kvclient/rangefeed/rangefeed.go:428 +0xe7
github.com/cockroachdb/cockroach/pkg/kv/kvclient/rangefeed.(*RangeFeed).run.func3({0x844f280?, 0xc010055ae0?})
	pkg/kv/kvclient/rangefeed/rangefeed.go:385 +0xf6
github.com/cockroachdb/cockroach/pkg/util/ctxgroup.GoAndWait.Group.GoCtx.func1()
	pkg/util/ctxgroup/ctxgroup.go:199 +0x8b
golang.org/x/sync/errgroup.(*Group).Go.func1()
	external/org_golang_x_sync/errgroup/errgroup.go:78 +0x50
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 1982602
	external/org_golang_x_sync/errgroup/errgroup.go:75 +0x96
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=16</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #149391 pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test: TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr failed [C-test-failure O-robot P-2 T-sql-foundations branch-master]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-52363

pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test.TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20268701?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20268701?buildTab=artifacts#/) on release-25.3 @ [a0b8e47cfb8ee55f4ad215a07a4fc7cca98d6b8f](https://github.com/cockroachdb/cockroach/commits/a0b8e47cfb8ee55f4ad215a07a4fc7cca98d6b8f):


```
          USE multiregion_db;
          SELECT crdb_internal.validate_multi_region_zone_configs()
          ----
    framework.go:559: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/11900/execroot/com_github_cockroachdb_cockroach/bazel-out/s390x-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:27:
          stage-exec phase=PostCommitNonRevertiblePhase stage=:
          USE multiregion_db;
          SELECT crdb_internal.validate_multi_region_zone_configs()
          ----
    panic.go:635: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/719c1f7a519ba65ba193fce224119da0/logTestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr1411188807
--- FAIL: TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr (215.95s)
=== RUN   TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_4_of_24
--- FAIL: TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_4_of_24 (97.75s)
=== RUN   TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_4_of_24/restore_all_tables_in_database
    sql_runner.go:117: 
        	Error Trace:	pkg/testutils/sqlutils/sql_runner.go:130
        	            				pkg/testutils/sqlutils/sql_runner.go:391
        	            				pkg/sql/schemachanger/sctest/framework.go:964
        	            				pkg/sql/schemachanger/sctest/backup.go:464
        	Error:      	Received unexpected error:
        	            	query 'SELECT status, job_type, description FROM [SHOW JOBS] WHERE job_type IN ('SCHEMA CHANGE', 'TYPEDESC SCHEMA CHANGE', 'NEW SCHEMA CHANGE') AND status NOT IN ('succeeded', 'failed')': expected:
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry.func1
        	            	  | 	pkg/testutils/sqlutils/sql_runner.go:394
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
        	            	  | 	pkg/testutils/soon.go:71
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
        	            	  | 	pkg/util/retry/retry.go:236
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
        	            	  | 	pkg/testutils/soon.go:77
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).succeedsWithin
        	            	  | 	pkg/testutils/sqlutils/sql_runner.go:130
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry
        	            	  | 	pkg/testutils/sqlutils/sql_runner.go:391
        	            	  | github.com/cockroachdb/cockroach/pkg/sql/schemachanger/sctest.waitForSchemaChangesToFinish
        	            	  | 	pkg/sql/schemachanger/sctest/framework.go:964
        	            	  | github.com/cockroachdb/cockroach/pkg/sql/schemachanger/sctest.exerciseBackupRestore.func1
        	            	  | 	pkg/sql/schemachanger/sctest/backup.go:464
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1690
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_s390x.s:797
        	            	Wraps: (2) query 'SELECT status, job_type, description FROM [SHOW JOBS] WHERE job_type IN ('SCHEMA CHANGE', 'TYPEDESC SCHEMA CHANGE', 'NEW SCHEMA CHANGE') AND status NOT IN ('succeeded', 'failed')': expected:
        	            	  |
        	            	  | got:
        	            	  | running, NEW SCHEMA CHANGE, ALTER TABLE multiregion_db.public.table_regional_by_row ADD … NG HASH
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
--- FAIL: TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_4_of_24/restore_all_tables_in_database (52.04s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #150802 pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test: TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-25.3.0-rc]
- #149391 pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test: TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr failed [C-test-failure O-robot P-2 T-sql-foundations branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/schemachanger/sctest/backup.go
+++ b/pkg/sql/schemachanger/sctest/backup.go
@@ -107,7 +107,7 @@ var runAllBackups = flag.Bool(
 	"if true, run all backups instead of a random subset",
 )
 
-const skipRate = .5
+const skipRate = .6
 
 func maybeRandomlySkip(t *testing.T) {
 	if !*runAllBackups && rand.Float64() < skipRate {

